pkg/testutils/moduletemplate.go

```diff
@@ -79,6 +80,31 @@ func UpdateModuleTemplateSpec(ctx context.Context,
 	return nil
 }
 
+func MandatoryModuleTemplateHasExpectedLabel(ctx context.Context, clnt client.Client, moduleName, key, value string,
+) error {
+	mandatoryModuleTemplates, err := templatelookup.GetMandatory(ctx, clnt)
+	if err != nil {
+		return err
+	}
+
+	var moduleTemplate *v1beta2.ModuleTemplate
+	for _, moduleTemplateInfo := range mandatoryModuleTemplates {
+		if moduleTemplateInfo.ModuleTemplate.Labels[shared.ModuleName] == moduleName {
+			moduleTemplate = moduleTemplateInfo.ModuleTemplate
+			break
+		}
+	}
+
+	if moduleTemplate == nil {
+		return fmt.Errorf("module template not found, %s", moduleName)
+	}
+
+	if moduleTemplate.Labels[key] != value {
+		return fmt.Errorf("label %s:%s not found", key, value)
+	}
+	return nil
+}
+
```

Comment:
I am okay with this as is, but one idea to simplify may be to use the lookup by label already. I.e., get all ModuleTemplates with `operator.kyma-project.io/mandatory-module: true`, assert that the len of returned ModuleTemplates is 1 and the entry has the expected module name.

Reply:
the GetMandatory already uses the label for the lookup

Reply:
Sorry, seems like I wasn't fully awake yet when reviewing this